src/libraries/System.Private.CoreLib/generators/EventDataBuilder.cs

```diff
@@ -0,0 +1,79 @@
+            {
+                StringBuilder _builder = builder;
+                int eventDataSize = events.Count + 1;
+
+                // check if any events ID went "out of sync" and is bigger than total # of events.
+                foreach (var esEvent in events)
+                {
+                    if (Int32.Parse(esEvent.Id) > eventDataSize)
+                    {
+                        eventDataSize = Int32.Parse(esEvent.Id) + 1;
+                    }
+                }
+
+                _builder.AppendLine("        m_EventDataInitializer = () => ");
+                _builder.AppendLine("        {");
+                
+
+                _builder.Append("            EventMetadata[] eventData = new EventMetadata[").Append(eventDataSize).AppendLine("];");
+                _builder.AppendLine("            eventData[0].Name = \"\";");
+
+                foreach (var esEvent in events)
+                {
+                    int task = (int)(0xFFFE - Int32.Parse(esEvent.Id));
+
+                    _builder.Append("            eventData[").Append(esEvent.Id).AppendLine("].Descriptor = new EventDescriptor(");
+                    _builder.Append("                ").Append(esEvent.Id).AppendLine(",");
+                    _builder.Append("                ").Append(esEvent.Version).AppendLine(",");
+                    _builder.AppendLine("                (byte)0,"); // TODO: ADD CHANNELS SUPPORT
+                    _builder.Append("                (byte)").Append(esEvent.Level).AppendLine(",");
+                    _builder.AppendLine("                (byte)0,"); // TODO: ADD OPCODE SUPPORT
+                    _builder.Append("                ").Append(task).AppendLine(",");
+
+                    ulong kwMask;
+                    if (esEvent.Keywords == "")
+                    {
+                        kwMask = 0;
+                    }
+                    else
+                    {
+                        kwMask = (ulong)(Int64.Parse(esEvent.Keywords));
+                    }
+                    ulong sessionMaskAllKeywords = (ulong)0x0fU << 44; // SessionMask.All.ToEventKeywords() in EventSource.cs
+                    _builder.Append("                unchecked((long)").Append(kwMask | sessionMaskAllKeywords).AppendLine("));");
+
+                    _builder.Append("            eventData[").Append(esEvent.Id).AppendLine("].Tags=EventTags.None;");
+                    _builder.Append("            eventData[").Append(esEvent.Id).Append("].Name=\"").Append(esEvent.Name).AppendLine("\";");
+                    _builder.Append("            eventData[").Append(esEvent.Id).AppendLine("].Parameters=null;");
+                    _builder.Append("            eventData[").Append(esEvent.Id).AppendLine("].Message=\"\";");
+                    _builder.Append("            eventData[").Append(esEvent.Id).AppendLine("].ActivityOptions=0;");
```

Comment:
The eventData looks like it doesn't yet capture all the information that reflection would have captured, which I assume means that we would lose functionality if it was enabled? If I understood accurately I think we need to either:
a) make sure that this new code will not be activated by default after the PR is checked in
b) add enough functionality so that behavior is identical when this code path is in use on the EventSources that make use of it.